### ISSUE TITLE
Audit the remaining fused kernels for the repeated-backward gap in the double-backward guard

### ISSUE THREAD

**Issue by david-stan** [JetBrains-Research/opaque#898]
Parent: #769

## Problem

Every fused kernel carries a guard rejecting second differentiation on its inner
`_XBackward` function — `cross_entropy.py:409`, `geglu.py:169,432`,
`rope_embedding.py:235,450`, `fused_add_rms_norm.py:390`,
`linear_cross_entropy.py:825`, `rms_norm.py:485`, `swiglu.py:170`, `lora.py` (x6),
`moe.py:129`, `fused_moe.py:324`, `_grouped_moe.py:229`.

That guard only fires when autograd differentiates *through* the backward pass.
Two ordinary first-order backwards over a retained graph never enter
`_XBackward.backward` at all, so the guard misses them:

```
second order (create_graph=True)              -> guard FIRED
re-traversal (retain_graph, 2 first-order)    -> guard did NOT fire
```

For any kernel whose backward writes into tensors saved by its forward, that
second traversal reads overwritten data and returns silently wrong gradients.
Triton stores go through raw pointers and skip autograd's version counter, so
nothing raises.

## Scope

#817 / #874 covers SwiGLU, GeGLU and the LoRA MLP. This issue is the sweep over
the rest. Preliminary reading, to be confirmed rather than trusted:

| Kernel | Backward writes into | Exposed? |
| --- | --- | --- |
| `swiglu.py`, `geglu.py`, `lora.py` | saved forward activations | yes — #874 |
| `rms_norm.py:413`, `fused_add_rms_norm.py:314` | incoming `dY` when `in_place` | **unclear** — different pattern, needs checking |
| `rope_embedding.py:203,412` | own freshly allocated buffers | no |
| `moe.py`, `fused_moe.py`, `_grouped_moe.py` | out-of-place | no |
| `cross_entropy.py:371,451` | cloned first (#402) | no |
| `linear_cross_entropy.py`, `_linear_ce_chunked.py` | recomputes, no mutation | no |

The `in_place` dY overwrite in the two RMSNorm kernels is the one worth real
attention: overwriting an incoming gradient is a different hazard from
overwriting a saved activation, and whether it is safe depends on whether that
gradient tensor can be shared with a sibling node.

`_linear_ce_chunked.py` also has no second-order guard at all, unlike every other
`autograd.Function` here. Its backward is plain differentiable torch rather than
Triton, so there is no corruption risk, but it contains `.detach()` calls that
would silently drop terms from a second-order derivative instead of raising.
Unverified.

## Acceptance criteria

- Each kernel above is confirmed exposed or not exposed to the re-traversal gap,
  by test rather than by reading.
- Exposed kernels reject a second traversal the way the existing guards reject
  second order — a `NotImplementedError` from the outer `backward`, not an
  allocation:
  ```python
  if getattr(ctx, "_opaque_backward_done", False):
      raise NotImplementedError("Repeated backward not supported for X")
  ctx._opaque_backward_done = True
  ```
- `_linear_ce_chunked.py` either gains the second-order guard or documents why it
  does not need one.
- A CUDA-marked test per exposed kernel covering `backward(retain_graph=True)`
  followed by a second backward.

## Notes

`ctx` persists across traversals, so the flag above fires exactly once on the
second pass — verified. Cost is zero: no allocation, no kernel changes.

Found while implementing #817.
